### PR TITLE
client/authority-discovery: Don't add own address to priority group

### PR DESCRIPTION
**Pull request by mxinden**
In the scenario of a validator publishing the address of its sentry node
to the DHT, said sentry node should not add its own Multiaddr to the
peerset "authority" priority group.

Related to 70cfeff.

### CODE REVIEW

File: client/authority-discovery/src/tests.rs [hunk 1]
```diff
@@ -461,3 +461,102 @@ fn dont_stop_polling_when_error_is_returned() {
...
+	let validator_public = validator_key_store
+		.write()
+		.sr25519_generate_new(key_types::AUTHORITY_DISCOVERY, None)
+		.unwrap();
+
+	let sentry_network: Arc<TestNetwork> = Arc::new(Default::default());
+
+	let sentry_multiaddr = {
+		let peer_id = sentry_network.local_peer_id();
+		let address: Multiaddr = "/ip6/2001:db8:0:0:0:0:0:2".parse().unwrap();
```
Comment (tomaka) [inline client/authority-discovery/src/tests.rs:480]:
```suggestion
		let address: Multiaddr = "/ip6/2001:db8:0:0:0:0:0:2/tcp/30333".parse().unwrap();
```

File: client/authority-discovery/src/tests.rs [hunk 2]
```diff
@@ -461,3 +461,102 @@ fn dont_stop_polling_when_error_is_returned() {
...
+
+		address.with(multiaddr::Protocol::P2p(
+			peer_id.into(),
+		))
+	};
+
+	// Address of some other sentry node of `validator`.
+	let random_multiaddr = {
+		let peer_id = PeerId::random();
+		let address: Multiaddr = "/ip6/2001:db8:0:0:0:0:0:1".parse().unwrap();
```
Comment (tomaka) [inline client/authority-discovery/src/tests.rs:490]:
```suggestion
		let address: Multiaddr = "/ip6/2001:db8:0:0:0:0:0:1/tcp/30333".parse().unwrap();
```

**Comment by mxinden** [comment]
@tomaka thanks for the review! Would you mind approving the pull request?

**Comment by mxinden** [comment]
@romanb yes, precisely.

**Comment by tomaka** [comment]
> Would you mind approving the pull request?

For what it's worth I went for "Comment" because I didn't want the PR to be merged before my comment has been seen/answered/addressed, but also didn't want to entirely block the PR being merged.

**Pull request merged**